scripts/dmaketgz

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+# docker-maketgz
+#
+#***************************************************************************
+#                                  _   _ ____  _
+#  Project                     ___| | | |  _ \| |
+#                             / __| | | | |_) | |
+#                            | (__| |_| |  _ <| |___
+#                             \___|\___/|_| \_\_____|
+#
+# Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
+#
+# This software is licensed as described in the file COPYING, which
+# you should have received as part of this distribution. The terms
+# are also available at https://curl.se/docs/copyright.html.
+#
+# You may opt to use, copy, modify, merge, publish, distribute and/or sell
+# copies of the Software, and permit persons to whom the Software is
+# furnished to do so, under the terms of the COPYING file.
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+# KIND, either express or implied.
+#
+# SPDX-License-Identifier: curl
+#
+###########################################################################
+
+version="${1:-}"
+
+if [ -z "$version" ]; then
+  echo "Specify a version number!"
+  exit
+fi
+
+make distclean
+docker build \
+       --build-arg SOURCE_DATE_EPOCH="$(date -u +%s)" \
+       --build-arg UID="$(id -u)" \
+       --build-arg GID="$(id -g)" \
+       -t curl/curl .
+
+run="run --rm -it -u $(id -u):$(id -g) -v $(pwd):/usr/src -w /usr/src curl/curl"
+
+# shellcheck disable=SC2086
+# "Double quote to prevent globbing and word splitting" on the $run use below
+docker $run autoreconf -fi
+# shellcheck disable=SC2086
+docker $run ./configure --without-ssl --without-libpsl
+# shellcheck disable=SC2086
+docker $run make -sj8
+# shellcheck disable=SC2086
+docker $run ./maketgz $version
```

Comment:
```suggestion
docker run --rm -it -u $(id -u):$(id -g) -v $(pwd):/usr/src -w /usr/src curl/curl sh -c "
  set -e
  autoreconf -fi
  ./configure --without-ssl --without-libpsl
  make -sj8
  ./maketgz $version"
```
This make it a bit more compact and avoids the shellcheck issue.

Reply:
Neat! As I was too quick to merge the first version, I took your version to  #13391